### PR TITLE
[`README.md`] Split the `Usage` section into 'Configuring' and 'Advanced Configuration'

### PR DESCRIPTION
**Pull request by darkred**
I suggest splitting the `Usage` section into `Configuring` and `Advanced Configuration`,
to make it more convenient for the user.
*Based on [clean-regex README.md](https://github.com/RunDevelopment/eslint-plugin-clean-regex/blob/master/README.md) and [markdown README.md](https://github.com/eslint/eslint-plugin-markdown/blob/main/README.md)*

PS. I'm glad you have joined forces! Great news!

### CODE REVIEW

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
Thank you for this PR @darkred! I like the idea of this but there are a few things that have to be changed.

First and foremost, `README.md` is partially generated. Please run `npm run update` and commit the changes.

I'll continue the review after the CI passes.

**Comment by darkred** [comment]
I just did it, thanks.

**Comment by darkred** [comment]
Did I mess up? Should I should revert? It seems so

**Comment by RunDevelopment** [comment]
Ahhh, how I love npm.... `npm install` isn't deterministic, so you installed the wrong versions of our dependencies. It's just one of those annoying npm things...

Yes, please revert the commit, then use `npm ci` to reinstall the dependencies, and then it should work.

**Comment by darkred** [comment]
I just did (revert, then `npm ci` and finally `nmp run install`) , 
and it now shows 58 changed .md files:
<details>
<summary>screenshot</summary>


</details>

Should I do the commit?

**Comment by RunDevelopment** [comment]
No, please don't commit.

> I am confused.

Oh yeah, me too.

I don't really understand why this happens to you. Please wait a moment while I look into this.

**Comment by RunDevelopment** [comment]
Nope, it's something on your end. After running `npm run update` on my end, this is what changed:

<details>

image

</details>

All as expected.

---

Looking at your diff screenshot more closely, it seems like our tooling has problems with lines. Our tooling assumes that all files use Unix line ends (LF). Is that the case on your end?

**Comment by RunDevelopment** [comment]
I just tested it locally on my machine and your problem is most likely caused by Windows line ends. 

I'm sorry, this is our fault. We didn't tell git to always checkout our code as LF. I'll fix that.

**Comment by RunDevelopment** [comment]
Alright, I just merged the fix. Please sync your fork to get the changes and then checkout this branch again.

**Comment by darkred** [comment]
I just pressed "Fetch Origin" in GitHub desktop, and there no commits to pull from my `patch-1` branch or `master`.
Is it something more to do?

I am asking because I just tried `npm test` and it completes ok.

**Comment by darkred** [comment]
And to clarify:

Initially(2 hours ago) I only used github.com to make the PR, i.e. modified the README.md, and made a PR for it, via my browser . 
In other words, initially, I didn't use GitHub Desktop at all, i.e. I didn't fork locally nor run any commands via CLI.

&nbsp;
After seeing the `CI / update` test was failing, 
only then I cloned my fork locally, and then I tried all these CLI commands you suggested.

&nbsp;
And if you look at the changes, they were not only whitespace, but actual changes all irrelevant to my PR:

<details>
<summary>screen capture</summary>


</details>

**Comment by RunDevelopment** [comment]
Don't worry, I'm not blaming you. All of this really shouldn't be so difficult.

> I just pressed "Fetch Origin" in GitHub Desktop

No, that's not enough. I synced and updated your PR branch for you so the CI passes for now. If you're interested in how to that, here's [a guide by GitHub](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork).

I'll review now.

File: README.md [hunk 1]
```diff
@@ -41,34 +41,48 @@ npm install --save-dev eslint eslint-plugin-regexp
 
 <!--DOCS_IGNORE_END-->
 
-## :book: Usage
+## :book: Configuring
```
Comment (RunDevelopment) [inline README.md:44]: I think that section should still be called "Usage".

```suggestion
## :book: Usage
```

File: README.md [hunk 2]
```diff
@@ -41,34 +41,48 @@ npm install --save-dev eslint eslint-plugin-regexp
...
 
-- `plugin:regexp/recommended` ... This is the recommended configuration for this plugin.
-  See [lib/configs/recommended.ts](https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/configs/recommended.ts) for details.
+```js
+// .eslintrc.js
+module.exports = {
+    "plugins": [
+        "regexp"
+    ],
+    rules: {
```
Comment (RunDevelopment) [inline README.md:80]: Let's keep the quoting consistent.

```suggestion
    "rules": {
```

File: README.md [hunk 3]
```diff
@@ -41,34 +41,48 @@ npm install --save-dev eslint eslint-plugin-regexp
...
+        // Override/add rules settings here, such as:
+        'regexp/rule-name': 'error'
```
Comment (RunDevelopment) [inline README.md:82]:
```suggestion
        "regexp/rule-name": "error"
```

File: README.md [hunk 4]
```diff
@@ -41,34 +41,48 @@ npm install --save-dev eslint eslint-plugin-regexp
...
 
-Create `.eslintrc.*` file to configure rules. See also: [http://eslint.org/docs/user-guide/configuring](http://eslint.org/docs/user-guide/configuring).
+Add `regexp` to the plugins section of your `.eslintrc` configuration file (you can omit the `eslint-plugin-` prefix)  
+and either use the recommended configuration or configure the rules you want:
 
-Example **.eslintrc.js**:
+### The recommended configuration
+
+This plugin provides one config: `regexp/recommended`, it enables a subset of core rules, and it's the recommended configuration.
+*See https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/configs/recommended.ts for details.*
```
Comment (RunDevelopment) [inline README.md:54]:
```suggestion
The `regexp/recommended` config enables a subset of [the core rules](#white_check_mark-rules) that should be most useful to most users.
*See https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/configs/recommended.ts for more details.*
```

File: README.md [hunk 5]
```diff
@@ -41,34 +41,48 @@ npm install --save-dev eslint eslint-plugin-regexp
...
+        "plugin:regexp/recommended"
+    ]
 }
 ```
 
-### Configuration
+### Advanced Configuration
 
-This plugin provides one config:
+Override/add specific rules settings. *See also: [http://eslint.org/docs/user-guide/configuring](http://eslint.org/docs/user-guide/configuring)*.
```
Comment (RunDevelopment) [inline README.md:72]: I have a problem with the word "settings". Settings are a different concept in ESLint, so we should use a different word here.

```suggestion
Override/add specific rules configurations. *See also: [http://eslint.org/docs/user-guide/configuring](http://eslint.org/docs/user-guide/configuring)*.
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

**Comment by darkred** [comment]
Thank you for the review. I committed all your suggested changes.

**Comment by RunDevelopment** [comment]
Thank you for contributing @darkred!

**Comment by darkred** [comment]
Glad I was able to contribute!

**Comment by darkred** [comment]
// offtopic

> All of this really shouldn't be so difficult.

https://twitter.com/natfriedman/status/1390350476009377795

> Working on a fix, and notice your fork is behind? Now, you can sync from the parent with just a single click in the GitHub UI. > 👇
> 
> Hope you like it! 🌟

Finally !! 😎👍

// offtopic

**Pull request merged**